src/NServiceBus.Core/Hosting/StartupDiagnostics/HostStartupDiagnosticsWriter.cs

```diff
@@ -58,6 +49,32 @@ await diagnosticsWriter(data)
             }
         }
 
+        List<StartupDiagnosticEntries.StartupDiagnosticEntry> DeduplicateEntries(List<StartupDiagnosticEntries.StartupDiagnosticEntry> entries)
+        {
+            var countMap = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+            var newList = new List<StartupDiagnosticEntries.StartupDiagnosticEntry>();
+            
+            foreach (var entry in entries)
+            {
+                if (!countMap.ContainsKey(entry.Name))
+                {
+                    countMap.Add(entry.Name, 1);
+                    newList.Add(entry);
+                }
+                else
+                {
```

Comment:
It may be worth logging (at `WARN`?) when this happens. Ideally this gets found and fixed.

Reply:
Done.